utils.py

```diff
@@ -177,59 +184,66 @@ def get_mistake_blunder_likelihood_from_fen(fen, mistake_threshold=200, blunder_
             if diff <= blunder_threshold * -1:
-                blunder_likelihood += get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci")        
+                blunder_likelihood += get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci")
             if diff > mistake_threshold * -1:
                 good_move.append(move)
         else:
             if diff >= mistake_threshold:
-                mistake_likelihood += get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci")    
+                mistake_likelihood += get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci")
             if diff >= blunder_threshold:
-                blunder_likelihood += get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci")        
+                blunder_likelihood += get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci")
             if diff < mistake_threshold:
                 good_move.append(move)
 
         if verbose==True:
-            print(move["san"], 
-                  get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci"), 
+            print(move["san"],
+                  get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci"),
                   total_games,
                   round(get_total_games_played(opening_explorer_from_fen, from_move=move["uci"], move_system="uci") / total_games, 3),
                   diff
              )
-    
+
     try:
         print("")
         print("Opponent has", len(good_move), "good move(s) in this position")
         print("Opponent has a", '{:.1%}'.format(mistake_likelihood / total_games), "chance to commit a mistake")
         print("Opponent has a", '{:.1%}'.format(blunder_likelihood / total_games), "chance to commit a blunder")
+
+        messages.append("")
+        messages.append("Opponent has " + str(len(good_move)) + " good move(s) in this position")
+        messages.append("Opponent has a " + '{:.1%}'.format(mistake_likelihood / total_games) + " chance to commit a mistake")
+        messages.append("Opponent has a " + '{:.1%}'.format(blunder_likelihood / total_games) + " chance to commit a blunder")
+
     except:
         print("Not enough games in Opening Explorer")
         print("")
-    
+
     return
 
 
-def get_sharpest_lines_from_fen(fen="rn1qk1nr/pp3pbp/4p1p1/2ppP3/3P4/2N2B1P/PPP2PP1/R1BQK2R w KQkq - 0 9", mistake_threshold=150, blunder_threshold=400, verbose=False):
-    # Put thresholds negative when analysing white moves 
+def get_sharpest_lines_from_fen(fen="rn1qk1nr/pp3pbp/4p1p1/2ppP3/3P4/2N2B1P/PPP2PP1/R1BQK2R w KQkq - 0 9", mistake_threshold=150, blunder_threshold=400, verbose=False, dev_limit=100):
```

Comment:
added a dev_limit=2 to speed up development.

Reply:
Great idea